// biga/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);

var styles = {"biga83__test":"_3vEMq","biga83__btn":"_3Kbua","biga83__btn_success":"_1pRRc","biga83__btn_info":"_ITa6t","biga83__btn_primary":"_3NNLr","biga83__btn_warning":"_xNTMX","biga83__btn_danger":"_a7hrW","biga83__dragdroplist":"_273Dd","biga83__dragdroplist__list":"_GJ9t9","biga83__dragdroplist__element":"_1MAY_","biga83__header":"_GPin1","biga83__input__form_container":"_36Mmo","biga83__input__form_icon":"_3910U","biga83__input__form":"_1xLr3","biga83__input__center":"_WgaXo","biga83__input__content_name":"_2WU6C","biga83__input__label_name":"_UoAOe","biga83__input__search_icon":"_1cyZc","biga83__input__search":"_3wpLh","biga83__info__modal":"_3R6Fh","biga83__info__modal_content":"_3w-Cd","biga83__info__modal_actions":"_3w88M","biga83__info__modal_backdrop":"_2TZs0","biga83__noti__container":"_3gCA7","biga83__noti__success":"_MEIT8","biga83__noti__danger":"_2itdm","biga83__noti__warning":"_1iHxJ","biga83__noti__info":"_2L8rn","biga83__noti__system":"_1jlRi","biga83__noti__default":"_3KwW4","biga83__noti__black":"_5xxW9","biga83__noti__showUpLeft":"_2jzx_","fadeinUpLeft":"_oN_8r","fadeoutUpLeft":"_1uRjS","biga83__noti__showUp":"_3Eiy2","fadeinUp":"_1zStL","fadeoutUp":"_13fu3","biga83__noti__showUpRight":"_JM6hU","fadeinUpRight":"_Jqu9t","fadeoutUpRight":"_2cv7e","biga83__noti__showRight":"_2TMYL","fadeinRight":"_yAHTv","fadeoutRight":"_2bmaG","biga83__noti__showDownRight":"_19oSi","fadeinDownRight":"_3Xhek","fadeoutDownRight":"_qirEz","biga83__noti__showDown":"_1IBpH","fadeinDown":"_8db5w","fadeoutDown":"_24bLM","biga83__noti__showDownLeft":"_2DNcU","fadeinDownLeft":"_2Uuqv","fadeoutDownLeft":"_1KFx9","biga83__noti__showLeft":"_3jFWw","fadeinLeft":"_2Gi-Q","fadeoutLeft":"_1QiQ9","biga83__noti__showCenter":"_2fj7M","fadeinCenter":"_2CDrw","fadeoutCenter":"_2f7gf","biga83__progrid__container":"_2_m95","biga83__progrid__header":"_XwkIS","biga83__progrid__main":"_2g3cM","biga83__progrid__footer":"_elB4u","biga83__progrid__title":"_W6JKi","biga83__progrid__title_fade":"_3QvPn","biga83__progrid__header_option":"_2pGkf","biga83__progrid__header_options":"_m2ovd","biga83__progrid__header_option_open":"_1zffn","biga83__progrid__header_search_segment":"_1My9D","biga83__progrid__header_search_box":"_1ojdv","biga83__progrid__header_search_text":"_1_Ig7","biga83__progrid__header_search_btn":"_OMxDY","biga83__progrid__modal":"_3-gEx","biga83__progrid__modal_content":"_WB6Nm","biga83__progrid__modal_actions":"_32s8B","biga83__progrid__modal_backdrop":"_1nkpq","biga83__simpletable__wrapper":"_22V9W","biga83__simpletable__table":"_3FoWl","biga83__simpletable__table_row":"_Ek74s","biga83__simpletable__table_header":"_13TBW","green":"_3axLA","blue":"_xHAXC","biga83__simpletable__table_row_cell":"__TVRv","biga83__table__container":"_1haJu","biga83__table__main":"_2kA77","biga83__table__div_button":"_2poBe","biga83__table__header":"_z_3yX","biga83__table__header_th_nodisplay":"_lk8SN","biga83__table__rows_tr":"_EksZo","biga83__table__rows_td_nodisplay":"_3rUm8","biga83__table__rows_td_options":"_3SYvw","biga83__table__rows_td_icon_red":"_3AGlp","biga83__table__rows_td_icon_blue":"_10VwX","biga83__table__rows_td_icon_green":"_3QFdt","biga83__table__modalrow":"_2-_Ag","biga83__table__modalrow_content":"_1Oh-T","biga83__table__modalrow_actions":"_3M_Is","biga83__table__modalrow_backdrop":"_WEVXI","biga83__table__pagination_footer":"_OClDh","biga83__table__pagination_ul":"_3_ifY","biga83__table__pagination_li_active":"_2Y7Ah","biga83__table__pagination_li_disabled":"_1UB8I","biga83__table__pagination_li_enabled":"_2y7u4","biga83__table__formfields":"_2N7l_","biga83__combobox":"_tx3JW","biga83__combobox__disabled":"_viz5v","biga83__combobox__disabled_cursor":"_hlCt5","biga83__combobox__options_container":"_3BR2r","biga83__combobox__selected":"_3cPaJ","biga83__combobox__selected_icon":"_2wBIz","biga83__combobox__active":"_1Brja","biga83__combobox__option":"_sUF1s","biga83__combobox__option_radio":"_1YhEb","biga83__combobox__searchbox":"_OupiD","biga83__spinner":"_27Uq9","spin":"_1sp4O","biga83__tabpanel":"_23KcY","biga83__tabpanel__header":"_9XA7W","biga83__tabpanel__active":"_2xGbV","biga83__tabpanel__indicator":"_1j_Y2","biga83__tabpanel__body":"_1hi-S","biga83__tabpanelvertical":"_28ZzB","biga83__tabpanelvertical__header":"_19t_G","biga83__tabpanelvertical__active":"_1OHFD","biga83__tabpanelvertical__body":"_1I-Ys","biga83__tabpanelvertical__indicator":"_3OTJU"};

var Button = function Button(_ref) {
  var kind = _ref.kind,
      title = _ref.title,
      type = _ref.type,
      click = _ref.click,
      style = _ref.style;
  var classes = [];
  var content = title;
  if (title.indexOf('icon') !== -1) content = /*#__PURE__*/React__default.createElement("i", {
    className: title.substr(5)
  });

  switch (kind) {
    case 'primary':
      classes.push(styles.biga83__btn, styles.biga83__btn_primary);
      break;

    case 'info':
      classes.push(styles.biga83__btn, styles.biga83__btn_info);
      break;

    case 'success':
      classes.push(styles.biga83__btn, styles.biga83__btn_success);
      break;

    case 'warning':
      classes.push(styles.biga83__btn, styles.biga83__btn_warning);
      break;

    case 'danger':
      classes.push(styles.biga83__btn, styles.biga83__btn_danger);
      break;

    default:
      classes.push(styles.biga83__btn);
      break;
  }

  return /*#__PURE__*/React__default.createElement("button", {
    type: type,
    className: classes.join(' '),
    style: style,
    onClick: click
  }, content);
};

var List = function List(props) {
  var drop = function drop(e) {
    e.preventDefault();
    var element_id = e.dataTransfer.getData('biga83__dad__element_id');
    var element = document.getElementById(element_id);
    e.target.appendChild(element);
  };

  var dragOver = function dragOver(e) {
    e.preventDefault();
  };

  return /*#__PURE__*/React__default.createElement("div", {
    id: props.id,
    onDrop: drop,
    onDragOver: dragOver,
    className: props.className
  }, props.children);
};

var ElementList = function ElementList(props) {
  var dragStart = function dragStart(e) {
    e.dataTransfer.setData('biga83__dad__element_id', e.target.id);
  };

  var dragOver = function dragOver(e) {
    e.stopPropagation();
  };

  return /*#__PURE__*/React__default.createElement("div", {
    id: props.id,
    className: props.className,
    draggable: props.draggable,
    onDragStart: dragStart,
    onDragOver: dragOver
  }, props.children);
};

var DragAndDropList = function DragAndDropList(_ref) {
  var dataInitial = _ref.dataInitial,
      dataFinal = _ref.dataFinal;
  var result1, result2;

  if (dataFinal !== null) {
    result1 = dataInitial.filter(function (d) {
      return !dataFinal.includes(d.id);
    });
    result2 = dataInitial.filter(function (d) {
      return dataFinal.includes(d.id);
    });
  }

  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__dragdroplist
  }, /*#__PURE__*/React__default.createElement(List, {
    id: "biga83__dad__list_1",
    className: styles.biga83__dragdroplist__list
  }, dataFinal !== null ? result1.map(function (d) {
    return /*#__PURE__*/React__default.createElement(ElementList, {
      id: d.id,
      key: d.id,
      className: styles.biga83__dragdroplist__element,
      draggable: "true"
    }, d.nombre);
  }) : dataInitial.map(function (d) {
    return /*#__PURE__*/React__default.createElement(ElementList, {
      id: d.id,
      key: d.id,
      className: styles.biga83__dragdroplist__element,
      draggable: "true"
    }, d.nombre);
  })), /*#__PURE__*/React__default.createElement(List, {
    id: "biga83__dad__list_2",
    className: styles.biga83__dragdroplist__list
  }, dataFinal !== null ? result2.map(function (d) {
    return /*#__PURE__*/React__default.createElement(ElementList, {
      id: d.id,
      key: d.id,
      className: styles.biga83__dragdroplist__element,
      draggable: "true"
    }, d.nombre);
  }) : null));
};

var HeaderTitle = function HeaderTitle(_ref) {
  var title = _ref.title,
      color = _ref.color;
  var colorTheme = {
    color: color
  };
  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__header,
    style: colorTheme
  }, /*#__PURE__*/React__default.createElement("h1", null, title));
};

var InputText = function InputText(_ref) {
  var onChangeInputHandler = _ref.onChangeInputHandler,
      title = _ref.title,
      icon = _ref.icon,
      value = _ref.value,
      type = _ref.type;
  return /*#__PURE__*/React__default.createElement("div", {
    className: type ? [styles.biga83__input__form_container, styles.biga83__input__center].join(' ') : styles.biga83__input__form_container
  }, /*#__PURE__*/React__default.createElement("div", {
    className: type ? styles.biga83__input__form : styles.biga83__input__search
  }, /*#__PURE__*/React__default.createElement("input", {
    type: "text",
    name: "name",
    autoComplete: "off",
    onChange: onChangeInputHandler,
    value: value,
    required: true
  }), /*#__PURE__*/React__default.createElement("label", {
    htmlFor: "name",
    className: styles.biga83__input__label_name
  }, /*#__PURE__*/React__default.createElement("span", {
    className: styles.biga83__input__content_name
  }, title))), icon ? /*#__PURE__*/React__default.createElement("div", {
    className: type ? styles.biga83__input__form_icon : styles.biga83__input__search_icon
  }, /*#__PURE__*/React__default.createElement("i", {
    className: icon
  })) : null);
};

var Modal = function Modal(props) {
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__info__modal_backdrop,
    onClick: props.onClose
  }), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__info__modal
  }, /*#__PURE__*/React__default.createElement("h3", null, props.title), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__info__modal_content
  }, props.children), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__info__modal_actions
  }, /*#__PURE__*/React__default.createElement(Button, {
    type: 'button',
    kind: 'info',
    click: props.onClose,
    title: 'icon fas fa-ban'
  }))));
};

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };
  return _extends.apply(this, arguments);
}

var ComboBox = function ComboBox(_ref) {
  var value = _ref.value,
      changed = _ref.changed,
      options = _ref.options;
  var backgroundTheme = {
    background: options.background
  };
  var borderTheme = {
    border: "8px solid " + (options.background === 'white' ? 'lightgray' : options.background)
  };

  var _useState = React.useState([]),
      dataCombo = _useState[0],
      setDataCombo = _useState[1];

  var _useState2 = React.useState(''),
      valueCombo = _useState2[0],
      setValueCombo = _useState2[1];

  var refCombobox = React.useRef(null);
  var refSelected = React.useRef(null);
  var refOptionsContainer = React.useRef(null);
  var refSearchbox = React.useRef(null);
  var refOptionsList = React.useRef(null);
  React.useEffect(function () {
    if (dataCombo.length === 0) {
      var key = null;
      var text;
      setDataCombo(options.data.map(function (cb, i) {
        text = [];
        Object.keys(cb).forEach(function (k) {
          if (options.key === k) key = cb[k];
          options.text.forEach(function (t) {
            if (t === k) text.push(cb[k]);
          });
        });
        return /*#__PURE__*/React__default.createElement("div", {
          key: options.title + key,
          className: styles.biga83__combobox__option
        }, /*#__PURE__*/React__default.createElement("input", {
          type: "radio",
          className: styles.biga83__combobox__option_radio,
          id: options.title + key,
          name: key
        }), /*#__PURE__*/React__default.createElement("label", {
          htmlFor: key
        }, text.join(' ')));
      }));
    }
  }, []);
  React.useEffect(function () {
    if (valueCombo !== '') {
      changed(options.title, valueCombo);
    }
  }, [valueCombo]);
  React.useEffect(function () {
    refOptionsContainer.current = refCombobox.current.children[0];
    refSelected.current = refCombobox.current.children[1];
    refSearchbox.current = refCombobox.current.children[2];
    refOptionsList.current = refOptionsContainer.current.childNodes;
    refOptionsList.current.forEach(function (o) {
      o.addEventListener("click", function () {
        setValueCombo(o.querySelector("label").htmlFor);
        refSelected.current.innerHTML = o.querySelector("label").innerHTML;
        refSelected.current.classList.add(styles.biga83__combobox__selected_icon);
        refOptionsContainer.current.classList.remove(styles.biga83__combobox__active);
      });
    });

    if (value) {
      refOptionsList.current.forEach(function (o) {
        if (value == o.querySelector("label").htmlFor) {
          setValueCombo(o.querySelector("label").htmlFor);
          refSelected.current.innerHTML = o.querySelector("label").innerHTML;
          refSelected.current.classList.add(styles.biga83__combobox__selected_icon);
          refOptionsContainer.current.classList.remove(styles.biga83__combobox__active);
        }
      });
    }

    if (dataCombo.length === 0 && value) {
      refSelected.current.innerHTML = value;
    }
  }, [dataCombo]);

  var displayOptions = function displayOptions() {
    refOptionsContainer.current.classList.toggle(styles.biga83__combobox__active);
    refSearchbox.current.value = "";
    filterList("");

    if (refOptionsContainer.current.classList.contains(styles.biga83__combobox__active)) {
      refSearchbox.current.focus();
    }
  };

  var filterList = function filterList(searchTerm) {
    searchTerm = searchTerm.toLowerCase();
    refOptionsList.current.forEach(function (option) {
      var label = option.firstElementChild.nextElementSibling.innerText.toLowerCase();

      if (label.indexOf(searchTerm) != -1) {
        option.style.display = "block";
      } else {
        option.style.display = "none";
      }
    });
  };

  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    className: options.disabled ? styles.biga83__combobox__disabled_cursor : null
  }, /*#__PURE__*/React__default.createElement("div", {
    ref: refCombobox,
    id: options.key,
    className: options.disabled ? [styles.biga83__combobox, styles.biga83__combobox__disabled].join(' ') : styles.biga83__combobox
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__combobox__options_container,
    style: backgroundTheme
  }, dataCombo), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__combobox__selected,
    onClick: displayOptions,
    value: valueCombo,
    style: backgroundTheme,
    onChange: function onChange(e) {
      return changed(e);
    }
  }, "Seleccionar ", options.title, /*#__PURE__*/React__default.createElement("i", {
    className: "fas fa-arrow-down"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__combobox__searchbox,
    onKeyUp: function onKeyUp(e) {
      return filterList(e.target.value);
    }
  }, /*#__PURE__*/React__default.createElement("input", {
    type: "text",
    placeholder: "Buscar...",
    style: borderTheme
  })))));
};

var MultiSelectCascade = function MultiSelectCascade(_ref) {
  var data = _ref.data,
      fields = _ref.fields,
      changed = _ref.changed,
      valuesMulti = _ref.valuesMulti;

  var _useState = React.useState(null),
      selects = _useState[0],
      setSelects = _useState[1];

  var _useState2 = React.useState({}),
      selectsData = _useState2[0],
      setSelectsData = _useState2[1];

  var _useState3 = React.useState({}),
      selectsValues = _useState3[0],
      setSelectsValues = _useState3[1];

  var changeValue = function changeValue(type, val) {
    changed(type.charAt(0).toLowerCase() + type.slice(1), val);
    fields.map(function (f, i) {
      if (f[0] === type.toLowerCase() && val !== selectsValues.values[type.toLowerCase()]) {
        var _extends2;

        setSelectsValues(_extends({}, selectsValues, {
          values: _extends({}, selectsValues.values, (_extends2 = {}, _extends2[type.toLowerCase()] = val, _extends2))
        }));

        if (fields.length > i + 1) {
          var _extends3;

          setSelects(null);
          var nextData = Array.from(new Set(data.filter(function (item) {
            if (item[fields[i][0]] === val) return true;else return false;
          }).map(function (item) {
            return item[fields[i + 1][0]];
          }))).map(function (o) {
            var _ref2;

            return _ref2 = {}, _ref2[fields[i + 1][0]] = o, _ref2;
          });
          setSelectsData(_extends({}, selectsData, {
            data: _extends({}, selectsData.data, (_extends3 = {}, _extends3[fields[i + 1][0]] = nextData, _extends3))
          }));
        } else {
          setSelects(null);
          setSelectsData(_extends({}, selectsData));
        }
      }
    });
  };

  React.useEffect(function () {
    var data = {};
    var values = {};
    fields.map(function (f, i) {
      if (valuesMulti) {
        values[f[0]] = valuesMulti[f[0]];
        data[f[0]] = null;
      } else {
        values[f[0]] = '';
        data[f[0]] = null;
      }
    });
    setSelectsData(_extends({}, selectsData, {
      data: data
    }));
    setSelectsValues(_extends({}, selectsValues, {
      values: values
    }));
  }, []);
  React.useEffect(function () {
    if (!(selectsData && Object.keys(selectsData).length === 0 && selectsData.constructor === Object)) {
      if (!selectsData.data[fields[0][0]]) {
        var _extends4;

        var newFirstData = Array.from(new Set(data.map(function (item) {
          return item[fields[0][0]];
        }))).map(function (o) {
          var _ref3;

          return _ref3 = {}, _ref3[fields[0][0]] = o, _ref3;
        });
        setSelectsData(_extends({}, selectsData, {
          data: _extends({}, selectsData.data, (_extends4 = {}, _extends4[fields[0][0]] = newFirstData, _extends4))
        }));
      }
    }
  }, [selectsData]);
  React.useEffect(function () {
    if (selectsData.data && selectsValues.values) {
      if (selectsData.data[fields[0][0]]) {
        setSelects(fields.map(function (f, i) {
          return /*#__PURE__*/React__default.createElement(ComboBox, {
            key: f[0],
            value: selectsValues.values[f[0]],
            changed: changeValue,
            options: {
              title: f[0].charAt(0).toUpperCase() + f[0].slice(1),
              data: selectsData.data[f[0]] ? selectsData.data[f[0]] : [],
              key: f[1][0],
              text: [f[1][0]],
              background: "white",
              disabled: f[0] === fields[0][0] ? false : selectsData.data[f[0]] ? false : true
            }
          });
        }));
      }
    }
  }, [selectsData]);
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, selects);
};

var Notification = function Notification(_ref) {
  var _ref$type = _ref.type,
      type = _ref$type === void 0 ? '' : _ref$type,
      _ref$message = _ref.message,
      message = _ref$message === void 0 ? 'Mensaje' : _ref$message,
      _ref$ubication = _ref.ubication,
      ubication = _ref$ubication === void 0 ? '' : _ref$ubication;
  var div = document.createElement('div');
  var notiClass = [];
  notiClass.push(styles.biga83__noti__container);

  switch (ubication) {
    case 'ul':
      notiClass.push(styles.biga83__noti__showUpLeft);
      break;

    case 'u':
      notiClass.push(styles.biga83__noti__showUp);
      break;

    case 'ur':
      notiClass.push(styles.biga83__noti__showUpRight);
      break;

    case 'r':
      notiClass.push(styles.biga83__noti__showRight);
      break;

    case 'dr':
      notiClass.push(styles.biga83__noti__showDownRight);
      break;

    case 'd':
      notiClass.push(styles.biga83__noti__showDown);
      break;

    case 'dl':
      notiClass.push(styles.biga83__noti__showDownLeft);
      break;

    case 'l':
      notiClass.push(styles.biga83__noti__showLeft);
      break;

    case 'c':
      notiClass.push(styles.biga83__noti__showCenter);
      break;

    default:
      notiClass.push(styles.biga83__noti__showCenter);
      break;
  }

  switch (type) {
    case 's':
      notiClass.push(styles.biga83__noti__success);
      break;

    case 'd':
      notiClass.push(styles.biga83__noti__danger);
      break;

    case 'w':
      notiClass.push(styles.biga83__noti__warning);
      break;

    case 'i':
      notiClass.push(styles.biga83__noti__info);
      break;

    case 'y':
      notiClass.push(styles.biga83__noti__system);
      break;

    case 'b':
      notiClass.push(styles.biga83__noti__black);
      break;

    default:
      notiClass.push(styles.biga83__noti__default);
      break;
  }

  div.className = notiClass.join(' ');
  var text = document.createTextNode(message);
  div.appendChild(text);
  document.body.appendChild(div);
  div.addEventListener('click', function () {
    div.className = div.className.replace(notiClass, '');
    div.parentNode.removeChild(div);
  });
  setTimeout(function () {
    if (document.querySelector('#biga83__noti__container') !== null) {
      div.className = div.className.replace(ubiClass, '');
      div.parentNode.removeChild(div);
    }
  }, 3000);
};

var SimpleTable = function SimpleTable(_ref) {
  var data = _ref.data;
  var keysRows = Object.keys(data[0]);
  var newHeadersRows = keysRows.map(function (h) {
    if (typeof h !== 'string') return '';
    return h.charAt(0).toUpperCase() + h.slice(1);
  });
  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__simpletable__wrapper
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__simpletable__table
  }, /*#__PURE__*/React__default.createElement("div", {
    className: [styles.biga83__simpletable__table_row, styles.biga83__simpletable__table_header].join(' ')
  }, newHeadersRows.map(function (h, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: styles.biga83__simpletable__table_row_cell,
      key: i
    }, h);
  })), data.map(function (r, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      className: styles.biga83__simpletable__table_row,
      key: i
    }, Object.values(r).map(function (cell, i) {
      return /*#__PURE__*/React__default.createElement("div", {
        className: styles.biga83__simpletable__table_row_cell,
        key: i,
        "data-title": Object.keys(r)[i].charAt(0).toUpperCase() + Object.keys(r)[i].slice(1)
      }, cell);
    }));
  })));
};

var Header = function Header(_ref) {
  var name = _ref.name,
      hidden = _ref.hidden;
  var classes = '';
  if (hidden.includes(name.charAt(0).toLowerCase() + name.slice(1))) classes = styles.biga83__table__header_th_nodisplay;
  return /*#__PURE__*/React__default.createElement("th", {
    className: classes
  }, name);
};

var TableHeader = function TableHeader(_ref) {
  var headerNames = _ref.headerNames,
      hidden = _ref.hidden,
      kind = _ref.kind;
  var newHeaders = headerNames.map(function (h) {
    if (typeof h !== 'string') return '';
    return h.charAt(0).toUpperCase() + h.slice(1);
  });
  return /*#__PURE__*/React__default.createElement("tr", {
    className: styles.biga83__table__header
  }, kind !== 's' ? kind !== 'i' ? /*#__PURE__*/React__default.createElement("th", null, "Acciones") : null : null, newHeaders.map(function (h) {
    return /*#__PURE__*/React__default.createElement(Header, {
      name: h,
      key: h,
      hidden: hidden
    });
  }));
};

var Rows = function Rows(_ref) {
  var type = _ref.type,
      row = _ref.row,
      extraData = _ref.extraData,
      keyId = _ref.keyId,
      kind = _ref.kind,
      hidden = _ref.hidden,
      updateItem = _ref.updateItem,
      deleteItem = _ref.deleteItem,
      formInputs = _ref.formInputs;

  var _useState = React.useState([]),
      tds = _useState[0],
      setTds = _useState[1];

  var values = Object.values(row);
  var keys = Object.keys(row);
  React.useEffect(function () {
    setTds(values.map(function (v, i) {
      if (hidden.includes(keys[i])) {
        return /*#__PURE__*/React__default.createElement("td", {
          className: styles.biga83__table__rows_td_nodisplay,
          style: {
            margin: '10px 0px'
          },
          key: i
        }, v);
      } else {
        var band = true;

        if (extraData.length > 0) {
          var value = null;
          extraData.forEach(function (e) {
            if (Object.keys(e).includes(keys[i])) {
              value = findValue(keys[i], v);
              band = false;
            }
          });

          if (value) {
            return /*#__PURE__*/React__default.createElement("td", {
              style: {
                margin: '10px 0px'
              },
              key: i
            }, value);
          }
        }

        if (band) {
          if (isValidDate(v)) {
            var _type = '';
            formInputs.forEach(function (f, j) {
              if (Object.keys(f)[0] === keys[i]) _type = Object.values(f)[0];
            });

            var _value = convertToDate(new Date(v), _type);

            return /*#__PURE__*/React__default.createElement("td", {
              style: {
                margin: '10px 0px'
              },
              key: i
            }, _value);
          } else {
            return /*#__PURE__*/React__default.createElement("td", {
              style: {
                margin: '10px 0px'
              },
              key: i
            }, v ? v : '');
          }
        }
      }
    }));
  }, [row]);

  var isValidDate = function isValidDate(d) {
    if (d && isNaN(d)) {
      if (new Date(d) instanceof Date && !isNaN(new Date(d))) return true;else return false;
    } else return false;
  };

  var redirectDetail = function redirectDetail(id) {
    if (id) history.push("/" + type + "/" + id, {
      id: id
    });
  };

  var options = [];

  if (kind.includes('f')) {
    options.push( /*#__PURE__*/React__default.createElement("i", {
      key: 1,
      className: ["fas fa-info", styles.biga83__table__rows_td_icon_green].join(' '),
      onClick: function onClick() {
        return redirectDetail(row.id);
      }
    }));
  }

  if (kind.includes('e')) {
    options.push( /*#__PURE__*/React__default.createElement("i", {
      key: 2,
      className: ["fas fa-edit", styles.biga83__table__rows_td_icon_blue].join(' '),
      onClick: function onClick() {
        return updateItem(type, row);
      }
    }));
  }

  if (kind.includes('d')) {
    options.push( /*#__PURE__*/React__default.createElement("i", {
      key: 3,
      className: ["fas fa-trash-alt", styles.biga83__table__rows_td_icon_red].join(' '),
      onClick: function onClick() {
        return deleteItem('d', row);
      }
    }));
  }

  var findValue = function findValue(key, value, index) {
    var exitValue = ' ';
    var band = false;
    formInputs.forEach(function (fi, i) {
      var valueInputs = Object.values(fi);
      var keyInputs = Object.keys(fi);

      if (Array.isArray(valueInputs[0])) {
        extraData.forEach(function (ed) {
          var valueExtra = Object.values(ed);
          var keyExtra = Object.keys(ed);

          if (keyExtra[0] === key && keyExtra[0] === keyInputs[0]) {
            band = true;
            valueExtra[0].forEach(function (ve, k) {
              var veValue = Object.values(ve);
              var veKey = Object.keys(ve);
              veKey.forEach(function (vv, vi) {
                if (vv === valueInputs[0][1]) {
                  if (veValue[vi] === value) {
                    valueInputs[0][2].forEach(function (va, va2) {
                      exitValue += ve[va] + " ";
                    });
                  }
                }
              });
            });
          }
        });
      }
    });
    if (!band) exitValue = value;
    return exitValue;
  };

  var convertToDate = function convertToDate(d, type) {
    var isMayor = function isMayor(d) {
      return d < 9 ? "0" + d : "" + d;
    };

    d.setTime(d.getTime() + d.getTimezoneOffset() * 60 * 1000);
    var date;

    switch (type) {
      case 'datetime':
        date = isMayor(d.getDate()) + "/" + isMayor(d.getMonth() + 1) + "/" + isMayor(d.getFullYear()) + "\n                        " + isMayor(d.getHours()) + ":" + isMayor(d.getMinutes()) + ":" + isMayor(d.getSeconds());
        return date;

      case 'date':
        date = isMayor(d.getDate()) + "/" + isMayor(d.getMonth() + 1) + "/" + isMayor(d.getFullYear());
        return date;
    }
  };

  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("tr", {
    key: keyId,
    className: styles.biga83__table__rows_tr
  }, kind !== 's' ? kind !== 'i' ? /*#__PURE__*/React__default.createElement("td", {
    className: styles.biga83__table__rows_td_options
  }, options) : null : null, tds));
};

var Rows$1 = React__default.memo(Rows);

var TableRows = function TableRows(_ref) {
  var type = _ref.type,
      data = _ref.data,
      extraData = _ref.extraData,
      kind = _ref.kind,
      hidden = _ref.hidden,
      updateItem = _ref.updateItem,
      deleteItem = _ref.deleteItem,
      formInputs = _ref.formInputs;
  return /*#__PURE__*/React__default.createElement("tbody", null, data.map(function (d, i) {
    return /*#__PURE__*/React__default.createElement(Rows$1, {
      type: type,
      row: d,
      extraData: extraData,
      key: i,
      kind: kind,
      hidden: hidden,
      updateItem: updateItem,
      deleteItem: deleteItem,
      formInputs: formInputs
    });
  }));
};

var Pagination = function Pagination(_ref) {
  var rowsCount = _ref.rowsCount,
      rowCount = _ref.rowCount,
      page = _ref.page,
      changePageHandler = _ref.changePageHandler;
  var pageShowLength = 5;
  var totalPages = Math.ceil(rowsCount / rowCount);
  var pages = [];

  if (totalPages >= 1) {
    var level = Math.ceil(page / pageShowLength) - 1;

    for (var i = 1; i <= pageShowLength; i++) {
      var showPage = i + pageShowLength * level;

      if (showPage <= totalPages) {
        if (showPage === page) pages.push( /*#__PURE__*/React__default.createElement("li", {
          className: styles.biga83__table__pagination_li_active,
          key: showPage,
          value: showPage,
          onClick: changePageHandler
        }, /*#__PURE__*/React__default.createElement("span", null, showPage)));else pages.push( /*#__PURE__*/React__default.createElement("li", {
          key: showPage,
          value: showPage,
          onClick: changePageHandler
        }, /*#__PURE__*/React__default.createElement("span", null, showPage)));
      }
    }
  }

  return /*#__PURE__*/React__default.createElement("footer", {
    className: styles.biga83__table__pagination_footer
  }, /*#__PURE__*/React__default.createElement("ul", {
    className: styles.biga83__table__pagination_ul
  }, /*#__PURE__*/React__default.createElement("li", {
    className: page === 1 ? ["pag__arrow-left", styles.biga83__table__pagination_li_disabled].join(' ') : ["pag__arrow-left", styles.biga83__table__pagination_li_enabled].join(' '),
    value: -1,
    onClick: changePageHandler
  }, /*#__PURE__*/React__default.createElement("i", {
    className: "fas fa-chevron-left"
  })), pages, /*#__PURE__*/React__default.createElement("li", {
    className: page === totalPages ? ["pag__arrow-right", styles.biga83__table__pagination_li_disabled].join(' ') : ["pag__arrow-right", styles.biga83__table__pagination_li_enabled].join(' '),
    value: 0,
    onClick: changePageHandler
  }, /*#__PURE__*/React__default.createElement("i", {
    className: "fas fa-chevron-right"
  }))));
};

var ModalRow = function ModalRow(props) {
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__modalrow_backdrop,
    onClick: props.onClose
  }), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__modalrow
  }, /*#__PURE__*/React__default.createElement("h3", null, props.title), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__modalrow_content
  }, props.children), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__modalrow_actions
  }, /*#__PURE__*/React__default.createElement(Button, {
    type: 'button',
    kind: 'info',
    click: props.onClose,
    title: 'icon fas fa-ban'
  }))));
};

var FormFields = function FormFields(_ref) {
  var data = _ref.data,
      extraData = _ref.extraData,
      formInputs = _ref.formInputs,
      hidden = _ref.hidden,
      required = _ref.required,
      disabled = _ref.disabled,
      getResult = _ref.getResult;

  var _useState = React.useState({}),
      result = _useState[0],
      setResult = _useState[1];

  var _useState2 = React.useState([]),
      inputs = _useState2[0],
      setInputs = _useState2[1];

  React.useEffect(function () {
    setResult(formInputs.reduce(function (obj, item) {
      return obj[Object.keys(item)] = '', obj;
    }, {}));
    if (data) setResult(_extends({}, data));
  }, []);
  React.useEffect(function () {
    if (Object.keys(result).length > 0 && result.constructor === Object) {
      setInputs(formInputs.map(function (f, i) {
        return chooseInput(Object.keys(f), Object.values(f), i);
      }));
    }
  }, [result]);

  var changeValues = function changeValues(e) {
    e.preventDefault();

    var newResult = _extends({}, result);

    newResult[e.target.name] = e.target.value;
    setResult(newResult);
  };

  var changeValueCombo = function changeValueCombo(type, val) {
    var newResult = _extends({}, result);

    newResult[type.charAt(0).toLowerCase() + type.slice(1)] = val;
    setResult(newResult);
  };

  var calculateDate = function calculateDate(val, type) {
    if (val) {
      var dateConvert = new Date(val);

      switch (type) {
        case 'date':
          dateConvert = dateConvert.toISOString().slice(0, 10);
          return dateConvert;

        case 'datetime':
          dateConvert = dateConvert.toISOString().slice(0, 16);
          return dateConvert;

        default:
          return null;
      }
    } else return null;
  };

  var chooseInput = function chooseInput(value, type, key) {
    var title = value[0].charAt(0).toUpperCase() + value[0].slice(1);
    var keys = Object.keys(result);
    var idValue = 0;
    keys.forEach(function (k, i) {
      if (k === value[0]) {
        idValue = i;
      }
    });
    var requiredOpt = {};
    if (required.includes(value[0])) requiredOpt["required"] = "required";

    if (hidden.includes(value[0])) {
      return /*#__PURE__*/React__default.createElement("input", {
        type: "hidden",
        key: key,
        name: value[0],
        value: Object.values(result)[idValue] || 0,
        onChange: changeValues
      });
    } else {
      switch (type[0]) {
        case 'number':
          return /*#__PURE__*/React__default.createElement("input", _extends({
            type: "number",
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: Object.values(result)[idValue] || 0,
            onChange: changeValues,
            disabled: disabled.includes(value[0]) ? true : false
          }));

        case 'string':
          return /*#__PURE__*/React__default.createElement("input", _extends({
            type: "text",
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: Object.values(result)[idValue] || "",
            onChange: changeValues,
            disabled: disabled.includes(value[0]) ? true : false
          }));

        case 'date':
          return /*#__PURE__*/React__default.createElement("input", _extends({
            type: "date",
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: calculateDate(Object.values(result)[idValue], 'date') || "",
            onChange: changeValues,
            disabled: disabled.includes(value[0]) ? true : false
          }));

        case 'datetime':
          return /*#__PURE__*/React__default.createElement("input", _extends({
            type: "datetime-local",
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: calculateDate(Object.values(result)[idValue], 'datetime') || "",
            onChange: changeValues,
            disabled: disabled.includes(value[0]) ? true : false
          }));

        case 'email':
          return /*#__PURE__*/React__default.createElement("input", _extends({
            type: "email",
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: Object.values(result)[idValue] || "",
            onChange: changeValues,
            disabled: disabled.includes(value[0]) ? true : false
          }));

        case 'textarea':
          return /*#__PURE__*/React__default.createElement("textarea", _extends({
            name: value[0],
            key: key,
            placeholder: title
          }, requiredOpt, {
            value: Object.values(result)[idValue] || "",
            disabled: disabled.includes(value[0]) ? true : false
          }));

        default:
          if (Array.isArray(type[0])) {
            switch (type[0][0]) {
              case 'select':
                return /*#__PURE__*/React__default.createElement(ComboBox, {
                  key: value[0],
                  value: Object.values(result)[idValue] || "",
                  changed: changeValueCombo,
                  options: {
                    title: value[0].charAt(0).toUpperCase() + value[0].slice(1),
                    data: extraData.find(function (e) {
                      return e[value[0]];
                    })[value[0]] || [],
                    key: type[0][1],
                    text: type[0][2],
                    background: "white",
                    disabled: disabled.includes(value[0]) ? true : false
                  }
                });

              case 'multiselectcascade':
                return /*#__PURE__*/React__default.createElement(MultiSelectCascade, {
                  key: value[0],
                  data: extraData.find(function (e) {
                    return e[value[0]];
                  })[value[0]] || [],
                  fields: type[0][1],
                  changed: changeValueCombo,
                  valuesMulti: data || null
                });

              default:
                return null;
            }
          }

      }
    }
  };

  var submitHandler = function submitHandler(e) {
    e.preventDefault();
    Object.keys(result).forEach(function (r, i) {
      if (r === hidden[0]) {
        if (Object.values(result)[i]) getResult('u', result);else getResult('a', result);
      }
    });
  };

  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__formfields
  }, /*#__PURE__*/React__default.createElement("form", {
    onSubmit: submitHandler
  }, inputs, /*#__PURE__*/React__default.createElement(Button, {
    kind: 'success',
    type: 'submit',
    title: "icon fas fa-save",
    style: {
      width: "100%"
    }
  })));
};

var Table = function Table(_ref) {
  var type = _ref.type,
      data = _ref.data,
      extraData = _ref.extraData,
      kind = _ref.kind,
      actionItem = _ref.actionItem,
      formInputs = _ref.formInputs,
      _ref$hidden = _ref.hidden,
      hidden = _ref$hidden === void 0 ? null : _ref$hidden,
      _ref$required = _ref.required,
      required = _ref$required === void 0 ? null : _ref$required,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? null : _ref$disabled,
      _ref$rowCount = _ref.rowCount,
      rowCount = _ref$rowCount === void 0 ? null : _ref$rowCount;

  var _useState = React.useState(''),
      filter = _useState[0],
      setFilter = _useState[1];

  var _useState2 = React.useState(1),
      actualPage = _useState2[0],
      setActualPage = _useState2[1];

  var _useState3 = React.useState([]),
      dataFinal = _useState3[0],
      setDataFinal = _useState3[1];

  var _useState4 = React.useState(0),
      totalRows = _useState4[0],
      setTotalRows = _useState4[1];

  var _useState5 = React.useState(false),
      show = _useState5[0],
      setShow = _useState5[1];

  var _useState6 = React.useState(''),
      titleModal = _useState6[0],
      setTitleModal = _useState6[1];

  var _useState7 = React.useState(null),
      fields = _useState7[0],
      setFields = _useState7[1];

  var loadData = React.useCallback(function (filter, actualPage, data) {
    setTotalRows(data.filter(function (c, i) {
      var newC = {};

      for (var prop in c) {
        if (!hidden.includes(prop)) Object.defineProperty(newC, prop, {
          enumerable: true,
          value: c[prop]
        });
      }

      return Object.values(newC).join(' ').toUpperCase().indexOf(filter.toUpperCase()) > -1;
    }).length);
    return data.filter(function (c, i) {
      var newC = {};

      for (var prop in c) {
        if (!hidden.includes(prop)) Object.defineProperty(newC, prop, {
          enumerable: true,
          value: c[prop]
        });
      }

      return Object.values(newC).join(' ').toUpperCase().indexOf(filter.toUpperCase()) > -1;
    }).filter(function (c, i) {
      return i >= actualPage * rowCount - rowCount && i <= actualPage * rowCount - 1;
    });
  }, []);
  React.useEffect(function () {
    setDataFinal(loadData(filter, actualPage, data));
  }, [filter, actualPage, data, loadData]);

  var filterInputHandler = function filterInputHandler(e) {
    setFilter(e.target.value);
    setActualPage(1);
    setDataFinal(loadData(filter, actualPage, data));
  };

  var prepareFields = function prepareFields(row) {
    setFields( /*#__PURE__*/React__default.createElement(FormFields, {
      data: row,
      extraData: extraData,
      formInputs: formInputs,
      hidden: hidden,
      required: required,
      disabled: disabled,
      getResult: getResult
    }));
  };

  var addHandler = function addHandler() {
    prepareFields(null);
    setShow(true);
    setTitleModal("Agregar " + type);
  };

  var updateHandler = function updateHandler(type, row) {
    prepareFields(row);
    setShow(true);
    setTitleModal("Editar " + type);
  };

  var closeModal = function closeModal() {
    setShow(false);
  };

  var changePageHandler = function changePageHandler(e) {
    switch (e.target.value) {
      case -1:
        var prevPage = actualPage - 1;
        setActualPage(prevPage);
        break;

      case 0:
        var nextPage = actualPage + 1;
        setActualPage(nextPage);
        break;

      default:
        setActualPage(e.target.value);
        break;
    }
  };

  var getResult = function getResult(type, data) {
    actionItem(type, data);
    closeModal();
  };

  var keys = Object.keys(data[0]);
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, show ? /*#__PURE__*/React__default.createElement(ModalRow, {
    title: titleModal,
    onClose: closeModal
  }, fields ? fields : /*#__PURE__*/React__default.createElement("p", null, "No hay nada para mostrar")) : null, /*#__PURE__*/React__default.createElement(InputText, {
    onChangeInputHandler: filterInputHandler,
    title: 'Buscar',
    icon: 'fas fa-search'
  }), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__container
  }, /*#__PURE__*/React__default.createElement("table", {
    className: styles.biga83__table__main
  }, /*#__PURE__*/React__default.createElement("thead", null, /*#__PURE__*/React__default.createElement(TableHeader, {
    headerNames: keys,
    hidden: hidden,
    kind: kind
  })), /*#__PURE__*/React__default.createElement(TableRows, {
    type: type,
    data: dataFinal,
    extraData: extraData,
    kind: kind,
    hidden: hidden,
    updateItem: updateHandler,
    deleteItem: actionItem,
    formInputs: formInputs
  }))), /*#__PURE__*/React__default.createElement(Pagination, {
    rowsCount: totalRows,
    rowCount: rowCount,
    page: actualPage,
    changePageHandler: changePageHandler
  }), kind.includes('i') ? /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__table__div_button
  }, /*#__PURE__*/React__default.createElement(Button, {
    kind: 'info',
    type: 'button',
    title: "icon fas fa-plus-circle",
    click: addHandler
  })) : null);
};

var Title = function Title(props) {
  var title = [];
  var _char = 0;
  var timer = null;

  var _useState = React.useState(null),
      mensaje = _useState[0],
      setMensaje = _useState[1];

  var refTitle = React.useRef(null);
  var refTitleChilds = React.useRef(null);
  React.useEffect(function () {
    title = props.title.split("");
    setMensaje(title.map(function (t, i) {
      return /*#__PURE__*/React__default.createElement("span", {
        key: i
      }, t);
    }));
  }, []);
  React.useEffect(function () {
    refTitleChilds.current = refTitle.current.childNodes;
    timer = setInterval(onTick, 50);
  }, [mensaje]);

  var onTick = function onTick() {
    refTitleChilds.current[_char].classList.add(styles.biga83__progrid__title_fade);

    _char++;

    if (_char === refTitleChilds.current.length) {
      complete();
      return;
    }
  };

  var complete = function complete() {
    clearInterval(timer);
    timer = null;
  };

  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("h3", {
    ref: refTitle,
    className: styles.biga83__progrid__title
  }, mensaje));
};

var Header$1 = function Header() {
  var _useState = React.useState([]),
      optionsClass = _useState[0],
      setOptionsClass = _useState[1];

  var _useState2 = React.useState(false),
      modalShow = _useState2[0],
      setModalShow = _useState2[1];

  React.useEffect(function () {
    setOptionsClass([].concat(optionsClass, [styles.biga83__progrid__header_option]));
  }, []);

  var showOptions = function showOptions(show) {
    if (show) setOptionsClass([].concat(optionsClass, [styles.biga83__progrid__header_option_open]));else setOptionsClass([optionsClass.filter(function (option) {
      return option !== styles.biga83__progrid__header_option_open;
    })]);
  };

  var configOptions = function configOptions() {
    console.log('show config');
    setModalShow(true);
  };

  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, modalShow ? /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__modal
  }, /*#__PURE__*/React__default.createElement("h3", null, "Holanda"), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__modal_content
  }, "hello"), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__modal_actions
  }, /*#__PURE__*/React__default.createElement(Button, {
    type: 'button',
    kind: 'info',
    click: function click() {
      return setModalShow(false);
    },
    title: 'icon fas fa-ban'
  }))) : null, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__header
  }, /*#__PURE__*/React__default.createElement("i", {
    onClick: function onClick() {
      return showOptions(true);
    },
    className: "fas fa-ellipsis-h"
  }), /*#__PURE__*/React__default.createElement("div", {
    className: optionsClass.join(' ')
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__header_options
  }, /*#__PURE__*/React__default.createElement("div", null, /*#__PURE__*/React__default.createElement("i", {
    onClick: function onClick() {
      return showOptions(false);
    },
    className: "fas fa-ellipsis-v"
  }), /*#__PURE__*/React__default.createElement("i", {
    onClick: configOptions,
    className: "fas fa-cog"
  })), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__header_search_segment
  }, /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__header_search_box
  }, /*#__PURE__*/React__default.createElement("input", {
    className: styles.biga83__progrid__header_search_text,
    type: "text",
    name: "",
    placeholder: "Buscar"
  }), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__header_search_btn
  }, /*#__PURE__*/React__default.createElement("i", {
    className: "fas fa-search"
  }))))))));
};

var ProGrid = function ProGrid(_ref) {
  var config = _ref.config;
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement(Title, {
    title: config.title
  }), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__container
  }, /*#__PURE__*/React__default.createElement(Header$1, null), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__main
  }, "main"), /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__progrid__footer
  }, "footer")));
};

var Spinner = function Spinner() {
  return /*#__PURE__*/React__default.createElement("div", {
    className: styles.biga83__spinner
  });
};

var TabPanel = function TabPanel(_ref) {
  var orientation = _ref.orientation,
      header = _ref.header,
      body = _ref.body;
  var refTabPanel = React.useRef(null);
  var refTabPanelHeader = React.useRef(null);
  var refTabPanelIndicator = React.useRef(null);
  var refTabPanelBody = React.useRef(null);
  var headerTabs = null;
  var bodyTabs = null;

  var renderOrientation = function renderOrientation() {
    refTabPanelHeader.current = refTabPanel.current.children[0];
    refTabPanelHeader.current.childNodes.forEach(function (h) {
      if (orientation === 'h') h.style.width = "calc(100% / " + body.length + ")";
    });
    refTabPanelIndicator.current = refTabPanel.current.children[1];
    refTabPanelBody.current = refTabPanel.current.children[2];
    if (orientation === 'h') refTabPanelIndicator.current.style.width = "calc(100% / " + header.length + ")";
    refTabPanelHeader.current.childNodes.forEach(function (o, i) {
      o.addEventListener("click", function () {
        refTabPanelHeader.current.childNodes.forEach(function (o2) {
          o2.classList.remove(orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active);
        });
        o.classList.add(orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active);
        refTabPanelBody.current.childNodes.forEach(function (o3) {
          o3.classList.remove(orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active);
        });
        refTabPanelBody.current.childNodes[i].classList.add(orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active);
        if (orientation === 'h') refTabPanelIndicator.current.style.left = "calc(calc(100% / " + header.length + ") * " + i + ")";else refTabPanelIndicator.current.style.top = "calc(80px + " + i * 50 + "px)";
      });
    });
  };

  headerTabs = header.map(function (h, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      key: h.title,
      className: i === 0 ? orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active : ""
    }, /*#__PURE__*/React__default.createElement("i", {
      className: h.icon
    }), " ", h.title);
  });
  bodyTabs = body.map(function (b, i) {
    return /*#__PURE__*/React__default.createElement("div", {
      key: i,
      className: i === 0 ? orientation === 'h' ? styles.biga83__tabpanel__active : styles.biga83__tabpanelvertical__active : ""
    }, orientation === 'v' ? b.icon ? /*#__PURE__*/React__default.createElement("i", {
      className: b.icon
    }) : null : null, b.title ? /*#__PURE__*/React__default.createElement("h2", null, b.title) : null, typeof b.content === "string" ? /*#__PURE__*/React__default.createElement("p", null, b.content) : b.content);
  });
  React.useEffect(function () {
    renderOrientation();
  }, []);
  return /*#__PURE__*/React__default.createElement(React__default.Fragment, null, /*#__PURE__*/React__default.createElement("div", {
    ref: refTabPanel,
    className: orientation === 'h' ? styles.biga83__tabpanel : styles.biga83__tabpanelvertical
  }, /*#__PURE__*/React__default.createElement("div", {
    className: orientation === 'h' ? styles.biga83__tabpanel__header : styles.biga83__tabpanelvertical__header
  }, headerTabs), /*#__PURE__*/React__default.createElement("div", {
    className: orientation === 'h' ? styles.biga83__tabpanel__indicator : styles.biga83__tabpanelvertical__indicator
  }), /*#__PURE__*/React__default.createElement("div", {
    className: orientation === 'h' ? styles.biga83__tabpanel__body : styles.biga83__tabpanelvertical__body
  }, bodyTabs)));
};

exports.Button = Button;
exports.ComboBox = ComboBox;
exports.DragAndDropList = DragAndDropList;
exports.HeaderTitle = HeaderTitle;
exports.InputText = InputText;
exports.Modal = Modal;
exports.MultiSelectCascade = MultiSelectCascade;
exports.Notification = Notification;
exports.ProGrid = ProGrid;
exports.SimpleTable = SimpleTable;
exports.Spinner = Spinner;
exports.TabPanel = TabPanel;
exports.Table = Table;
